File: Scripts/Python/Aug_27/laguerre.py
#!/usr/bin/env python
# File: laguerre.py
# Name: D.Saravanan
# Date: 27/08/2021

""" Script to calculate and plot Laguerre polynomials """

import numpy as np
import matplotlib
matplotlib.rcParams['pgf.texsystem'] = 'pdflatex'
matplotlib.rcParams.update({'font.family': 'serif', 'font.size': 8,
    'axes.labelsize': 10, 'axes.titlesize': 10, 'figure.titlesize': 10})
matplotlib.rcParams['text.usetex'] = True
import matplotlib.pyplot as plt
matplotlib.use('Agg')

fig, ax = plt.subplots()

for n in range(5):

    # generate the Laguerre basis for the polynomial of degree 'n'
    Ln = np.polynomial.laguerre.Laguerre.basis(n)

    # interval [-4, 7]
    x = np.linspace(-4, 7, 1000)
    y = Ln(x)

    #x, y = np.polynomial.Laguerre.basis(n, [-4,7]).linspace(1000)
    #x, y = np.polynomial.laguerre.Laguerre.basis(n, [-4,7]).linspace(1000)

    ax.plot(x, y, label=r'$L_{}(x)$'.format(n))
    
ax.set(xlabel=r'$x$', ylabel=r'$L_{n}(x)$', xlim=(-2,5), ylim=(-4,4))
ax.set_title(r'Laguerre polynomials')
ax.grid(); ax.legend()
plt.savefig('laguerre.pdf')

File: Scripts/Julia/Aug_27/laguerre.jl
#!/usr/bin/env julia
# File: laguerre.jl
# Name: D.Saravanan
# Date: 27/08/2021

""" Script to calculate and plot Laguerre polynomials """

import PyPlot
const plt = PyPlot
plt.rc("pgf", texsystem="pdflatex")
plt.rc("font", family="serif", weight="normal", size=8)
plt.rc("axes", labelsize=10, titlesize=10)
plt.rc("figure", titlesize=10)
plt.rc("text", usetex="True")
using LaTeXStrings

function Laguerre(x, n)
    n == 0 && return 1
    n == 1 && return 1-x
    ((2(n-1) + 1 - x)*Laguerre(x, n-1) - (n-1)*Laguerre(x, n-2))/n
end

fig, ax = plt.subplots()

for n in 0:4

    # interval [-4, 7]
    x = LinRange(-4, 7, 1000)
    y = map(x -> Laguerre(x, n), x)

    ax.plot(x, y, label=L"L_{n}(x)")
end

ax.set(xlabel=raw"$x$", ylabel=raw"$L_{n}(x)$", xlim=(-2,5), ylim=(-4,4))
ax.set_title(raw"Laguerre polynomials")
ax.grid(true); ax.legend()
plt.savefig("laguerre.png")
